### ISSUE TITLE
Final changes for 6.3

### ISSUE THREAD

**Issue by peastman** [pandegroup/openmm#992]
We're getting close to the 6.3 release.  We're well past the beta release at this point, so no new features are going in, but what bug fixes or cleanup still needs to be done?  Please post anything here so we'll have it all in one place.

**Comment by jchodera** [comment]
Why don't we use the GitHub issue labeling system?  You can collect up everything you have labeled `bug` and then assign them to Milestones: either `openmm 6.3` or `openmm 6.4`, etc.  That way, we can more easily point out anything we might have missed.

If we *don't* do this, we risk forgetting to do important things before release, like #503.

**Comment by peastman** [comment]
I'd prefer to use a single issue for this purpose to have everything in one place.  So if there's anything else you think should be none (whether or not there is an open issue about it on Github), please post it here.

**Comment by jchodera** [comment]
There is no reason we can't do both!  We have already missed important things we were supposed to include because we did not use a strategy of labeling issues with release targets.  We don't want to continue to miss things that may not be the most recent issues but are still important.

**Comment by leeping** [comment]
We could add the AMBER-FB15 parameters (https://github.com/pandegroup/openmm/pull/1009) since it's just one file.  Since this is so last-minute, I don't mind if we hold off on it - but I'll ask @VijayPande to weigh in.  :)

**Comment by peastman** [comment]
We should definitely wait till post-6.3 for this.  We're several weeks past the beta at this point, and well into the final bug fixes stage.  Not the time to be adding new features!

**Comment by informatorius** [comment]
I think performance issues should be fixed
https://github.com/pandegroup/openmm/issues/969
https://github.com/pandegroup/openmm/issues/949

**Comment by peastman** [comment]
#949 is just a matter of compiling against the right library.  #969 I'm really stumped by.  I've written to Nvidia about it, but haven't gotten any reply.

**Comment by jchodera** [comment]
Would still be helpful to tag issues with `OpenMM 6.3 release` explicitly so that everybody knows what will be fixed!  :)

**Comment by peastman** [comment]
I believe the Nvidia performance issue #969 is the only thing we're still waiting on.  Aside from that, there are *no* other changes waiting to be made before we release.  Correct?  Speak now or forever hold your peace!

**Comment by rmcgibbo** [comment]
+1, unless the peformance issue takes a long time. If, hypothetically, it were going to take 3 months before they release a fixed driver, then I would say we should just go back to committing as normal, and not hold off on changes.

**Comment by jchodera** [comment]
> Correct? Speak now or forever hold your peace!

I will plead once more to sweep through the issues and mark everything with a **Milestone** so that we are actually certain there are no issues we are forgetting for 6.3.

**Comment by peastman** [comment]
I've already gone through the issues.  Anything has not been mentioned here is something I didn't think needed to be fixed in 6.3, and apparently neither did anyone else!

**Comment by jchodera** [comment]
I've just looked through open issues, and I see lots of things that would be nice to have but no showstoppers.

**Comment by jchodera** [comment]
@peastman : I'm looking at your [changes to the OpenMM linux build environment](https://github.com/pandegroup/openmm/wiki/OpenMM-build-platforms-and-environments), but you don't specify which version of the AMD APP SDK you want.  Which version?  2.9, 2.9.1, or 3.0beta?

**Comment by peastman** [comment]
Let's do 2.9.1.  That's what I used for testing the performance.  And building against a beta sounds risky.

**Comment by jchodera** [comment]
I think I have the new VM properly configured. Should I rebuild a new beta?

**Comment by peastman** [comment]
Now that #1033 is merged, I believe we are ready to create the release candidate.  The current revision is 23fa9d95383e07e9c32dc283bde991d1f5fba176.  Unless someone objects, that's what it should be built from.  I'll go ahead and do the Windows build.

**Comment by rmcgibbo** [comment]
Shall we tag the release on GH too? https://github.com/pandegroup/openmm/releases/new

**Comment by jchodera** [comment]
+1 for tagging it as `6.3.0rc1`

**Comment by jchodera** [comment]
Whoops!  @rmcgibbo: Did your docs build OK in the provisioned VM?  I just noticed my packaging build ran into problems with TeX:
```
Running LaTeX files through pdflatex...
gmake[5]: execvp: pdflatex: Permission denied
gmake[5]: *** [OpenMMDeveloperGuide.pdf] Error 127
gmake[4]: *** [latexpdf] Error 2
make[3]: *** [sphinx-docs/developerguide/latex/OpenMMDeveloperGuide.pdf] Error 2
```
I think this is because the VM provisioning must have had some failures to install LaTeX packages.  Reprovisioning now.

**Comment by jchodera** [comment]
OK, I think I've got this sorted now:
* [`source`](https://jenkins.choderalab.org/job/package-openmm-source/ws/packaging/)
* [`osx`](https://jenkins.choderalab.org/job/package-openmm-osx/ws/packaging/)
* [`linux`](https://jenkins.choderalab.org/job/package-openmm-linux-vagrant/ws/packaging/)

Again, sorry this took so long.  VM provisioning took forever.  I'm working on upgrading our main build system.

**Comment by peastman** [comment]
The RC is now posted on the download page.  Give it a try!  I'll also send out an email to the beta testers mailing list.

**Comment by peastman** [comment]
I found a problem in the documentation.  Look, for example, at the very last sentence of section 4.3 in the user guide.  It starts out, "So in Example,Gromacs example we actually could have omitted this parameter...", whereas it ought to say, "So in Example 4-3 we actually could have omitted this parameter..."

Did you change the version of Sphinx installed on the virtual machine?  I've been investigating and it looks like a recent update to Sphinx broke the extension we use for autonumbering examples, figures, etc.  I'm trying to figure out how to fix it, but in the mean time I think we should just rebuild the docs with the older version of Sphinx and swap those into the packages.

**Comment by peastman** [comment]
I've posted new packages with rebuilt documentation.

**Comment by peastman** [comment]
We need to upload the new documentation to openmm.org (but not actually update the links on the documentation page until the release goes live).  I don't know how that site is managed.  Who is able to do it?

**Comment by rmcgibbo** [comment]
I'll do it. For future reference (it's probably good if you know), the site is a static bucket on AWS S3. I'll give you the account name/password in person.

**Comment by peastman** [comment]
Make sure you build the manuals with Sphinx 1.2.  There's a change in 1.3 that breaks one of the extensions we use.  (The fix is easy, and I'll check it in once 6.3 is out.)

**Comment by rmcgibbo** [comment]
Uploaded the sphinx http://docs.openmm.org/6.3.0/developerguide/, and http://docs.openmm.org/6.3.0/userguide/, with sphinx 1.2.3. When I get back from lunch I'll try to check the references and table numbering and stuff.

**Comment by peastman** [comment]
I checked a few spots and it looks fine.  Thanks!  We also need the API docs.  You can get those out of one of the installers.

**Comment by rmcgibbo** [comment]
API docs up now too using the docs from the installers. http://docs.openmm.org/6.3.0/api-c++/annotated.html, http://docs.openmm.org/6.3.0/api-python/annotated.html.

The command I used is essentially `s3cmd put -r --no-mime-magic --guess-mime-type * s3://docs.openmm.org/6.3.0/<PATH>/`, for the various directories.

**Comment by rmcgibbo** [comment]
Also, for the next release, we might consider reformatting the sphinx docs in such a way that the landing page doesn't hit you with a big block of legal text. It's a little offputting.

**Comment by jchodera** [comment]
> Also, for the next release, we might consider reformatting the sphinx docs in such a way that the landing page doesn't hit you with a big block of legal text. It's a little offputting.

+1.  Create a separate issue?

**Comment by jchodera** [comment]
> The command I used is essentially s3cmd put -r --no-mime-magic --guess-mime-type * s3://docs.openmm.org/6.3.0/<PATH>/, for the various directories.

Let's try to capture some of that knowledge here:
https://github.com/pandegroup/openmm/wiki/Cutting-an-OpenMM-release

I've taken a stab at starting this.

**Comment by rmcgibbo** [comment]
Okay, I created #1047. Having a wiki space to put that stuff seems like a good idea.

**Comment by jchodera** [comment]
Thanks!

**Comment by peastman** [comment]
Are we ready to release 6.3?  If everyone thinks we're ready, I'll make it live on simtk.org and post the announcement.

**Comment by rmcgibbo** [comment]
Yeah, +1. Also tag on GH.

**Comment by peastman** [comment]
Right.  Or rather, create a full release, not just a tag.

**Comment by jchodera** [comment]
+1

**Comment by peastman** [comment]
The github release is created, and the download at simtk.org is public.  That means 6.3 is now officially released!

I'll also post an announcement to the forum.  Other things that need to be done are releasing the conda packages, and updating the links to documentation at openmm.org.

**Comment by kyleabeauchamp** [comment]
Nicely done guys

**Comment by jchodera** [comment]
The simtk.org page says the release date was June 8, 2015:
https://simtk.org/project/xml/downloads.xml?group_id=161

Do you maybe want to update that with the actual release date, not the beta date?

**Comment by peastman** [comment]
Done.  Good catch.

**Comment by jchodera** [comment]
Conda packages for `linux` and `osx` releases are up:
https://binstar.org/omnia/openmm/files

Looks like we're still having trouble with appveyor building `win-32`.  Trying to figure out why.

**Comment by rmcgibbo** [comment]
No win-32 CUDA 7.0 libraries. Also the recipe probably isn't right for win.

**Comment by peastman** [comment]
The documentation links still need to be updated.  Who manages the openmm.org site?  I don't even know where it's hosted.

**Comment by rmcgibbo** [comment]
> Who manages the openmm.org site?

Nobody really. I guess yutong and me. I'll do the links. I should transfer the keys to you.

**Comment by peastman** [comment]
We should make sure there are several people who can maintain the site.  Definitely both me and Joy, and probably a few other people as well.  I'll bring it up at the next OpenMM meeting.

**Comment by rmcgibbo** [comment]
Okay. It's all here: https://github.com/pandegroup/openmm-org/

-Robert

**Comment by rmcgibbo** [comment]
Links are updated at http://openmm.org/documentation.html. The pages can be
updated just by modifying the html and committing to master or merging a pr
into https://github.com/pandegroup/openmm-org, and then travis syncs them
to amazon.

**Comment by rmcgibbo** [comment]
We could also update the benchmarks at
http://wiki.simtk.org/openmm/BenchmarkOpenMMDHFR /
http://openmm.org/about.html#benchmarks

**Comment by peastman** [comment]
Running benchmarks is one of the next items on my list.  I've been trying to get hold of a newer AMD GPU for running benchmarks on.  The 7970 I've been using is now two generations out of date, so it's not a fair comparison to put it next to benchmarks from current generation Nvidia GPUs.

**Comment by jchodera** [comment]
Should we create a separate issue thread about running and posting benchmarks?

**Comment by peastman** [comment]
I've already posted the Nvidia and CPU benchmarks.  We have an R9 Fury X on order, and I'll run benchmarks on that as soon as it comes.

**Comment by jchodera** [comment]
The landing page benchmarks haven't been updated yet:
http://openmm.org/about.html#benchmarks

Do you want me to submit a PR for that?

**Comment by peastman** [comment]
Sure.  There will also be further updates to the benchmarks once the new AMD GPU arrives.

**Comment by peastman** [comment]
The AMD benchmarks are now up at http://wiki.simtk.org/openmm/BenchmarkOpenMMDHFR.

**Comment by kyleabeauchamp** [comment]
Might be good to add the OS version?

**Comment by peastman** [comment]
@jchodera still waiting for you to update the benchmarks at openmm.org.

**Comment by jchodera** [comment]
Whoops!  My bad:

https://github.com/pandegroup/openmm-org/pull/4
